tests/mysql/query.rs

```diff
@@ -899,6 +899,54 @@ fn select_53() {
     );
 }
 
+#[test]
+fn select_54() {
+    assert_eq!(
+        Query::select()
+            .columns(vec![Glyph::Aspect,])
+            .from(Glyph::Table)
+            .and_where(Expr::expr(Expr::col(Glyph::Aspect).if_null(0)).gt(2))
+            .order_by(
+                Glyph::Id,
+                Order::Field(Values(vec![4.into(), 5.into(), 1.into(), 3.into(),]))
+            )
+            .order_by((Glyph::Table, Glyph::Aspect), Order::Asc)
+            .to_string(MysqlQueryBuilder),
+        [
+            r#"SELECT `aspect`"#,
+            r#"FROM `glyph`"#,
+            r#"WHERE IFNULL(`aspect`, 0) > 2"#,
+            r#"ORDER BY FIELD(`id`, 4, 5, 1, 3),"#,
+            r#"`glyph`.`aspect` ASC"#,
+        ]
+        .join(" ")
+    );
+}
+
+#[test]
+fn select_55() {
+    assert_eq!(
+        Query::select()
+            .columns(vec![Glyph::Aspect,])
+            .from(Glyph::Table)
+            .and_where(Expr::expr(Expr::col(Glyph::Aspect).if_null(0)).gt(2))
+            .order_by((Glyph::Table, Glyph::Aspect), Order::Asc)
+            .order_by(
+                Glyph::Id,
+                Order::Field(Values(vec![4.into(), 5.into(), 1.into(), 3.into(),]))
+            )
+            .to_string(MysqlQueryBuilder),
+        [
+            r#"SELECT `aspect`"#,
+            r#"FROM `glyph`"#,
+            r#"WHERE IFNULL(`aspect`, 0) > 2"#,
+            r#"ORDER BY `glyph`.`aspect` ASC,"#,
+            r#"FIELD(`id`, 4, 5, 1, 3)"#,
```

Comment:
According to the [MySQL docs of `FIELD` function](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_field)

> If str is NULL, the return value is 0 because NULL fails equality comparison with any value. FIELD() is the complement of ELT().

That mean if the value of `id` column is not exists in array `[4, 5, 1, 3]`, then those rows will be sorted and placed in front of those with `id` exists in the array.

This behaviour is different from PostgreSQL & SQLite. Which both of them placed those rows behind instead of in the front.

Reply:
You are right. I will modify the `MySQL` query builder to produce:
```sql
SELECT `aspect` FROM `glyph`
ORDER BY IF(`id` NOT IN (4, 5, 1, 3),4,0), FIELD(`id`, 4, 5, 1, 3)
```
This causes all the ids not in the set passed to the `FIELD` function to have an order or `4` (in this case as the length of the set is 4) and thus being placed after `4,5,1,3`.

Reply:
Or we use [case when](https://dev.mysql.com/doc/refman/8.0/en/case.html) syntax on all three databases?

Reply:
Oh, I see. Didn't know that it was supported by MySQL. Let me change it.